### ISSUE TITLE
`examples/download_user.py` not downloading the last page

### ISSUE THREAD

**Issue by acctfpn** [scrazzz/redgifs#23]
### Summary

`examples/download_user.py` not downloading the last page

### Reproduction Steps

Download one user with it.

### Expected Result

All files are downloaded.

### Actual Result

It downloads n - 1 pages

### System Information

> python -m redgifs --version
- Python v3.11.6-final
- redgifs v1.7.2-final
- aiohttp v3.8.6
- system info: Linux 6.5.8-200.fc38.x86_64 #1 SMP PREEMPT_DYNAMIC Fri Oct 20 15:53:48 UTC 2023

### Checklist

- [X] I have searched all open and closed issues for any duplicates.

### Additional Information

This line is the problem: [`while current_page != total_pages:`](https://github.com/scrazzz/redgifs/blame/d784b6666b5cb09618c6779924a77ee8d0819551/examples/download_user.py#L20)

I'm using something like this.
```py
    while True:
...
        if current_page <= total_pages:
            # Clear the old gifs from the previous page
            current_page_gifs.clear()
            # Make a new API call to get the gifs from the next page
            data = search(search_term, page=current_page)
            # Update `current_page_gifs` with the new gifs
            current_page_gifs.extend(data.gifs)
        else:
            break
```

**Comment by scrazzz** [comment]
Good catch, thanks for pointing out. I have fixed it (hopefully): 179a46d798358cec4a29d0395f6a20c635dfb7b4.

**Comment by acctfpn** [comment]
That fixed this issue but created another one.
Now it searches  total_pages + 1 and throws an exception because the last page does not exist.
The `if current_page <= total_pages:` check has to be right before actually searching like in the script above.

**Comment by scrazzz** [comment]
Yeah you're right. I'll get back to it when I get the time.

**Comment by scrazzz** [comment]
This issue has been fixed (72db39aa04253e74dff163859575fa97548a0918)! Thanks @acctfpn for filing this issue.
